static/app/components/performanceDuration.tsx

```diff
@@ -0,0 +1,36 @@
+import Duration from 'sentry/components/duration';
+import {defined} from 'sentry/utils';
+
+interface DurationProps {
+  abbreviation?: boolean;
+}
+
+interface SecondsProps extends DurationProps {
+  seconds: number;
+}
+
+interface MillisecondsProps extends DurationProps {
+  milliseconds: number;
+}
+
+type PerformanceDurationProps = SecondsProps | MillisecondsProps;
+
+function hasMilliseconds(props: PerformanceDurationProps): props is MillisecondsProps {
+  return defined((props as MillisecondsProps).milliseconds);
+}
+
+function PerformanceDuration(props: PerformanceDurationProps) {
+  const normalizedSeconds = hasMilliseconds(props)
+    ? props.milliseconds / 1000
+    : props.seconds;
+
+  return (
+    <Duration
```

Comment:
@Zylphrex what if we added the functionality that we need to the duration component and just remove the PerformanceDuration? Also, I think we should extend the API where instead of seconds and milliseconds. 

Maybe something like <Duration time={time} unit="s|ms|ns|"/> and then based on the unit, we normalize the data and format it to smallest meaningful number

Reply:
Ah, I see. Ok then 👍🏼